x-pack/plugins/security_solution/public/detection_engine/rule_creation/components/step_rule_actions/notification_action.tsx

```diff
@@ -91,13 +98,19 @@ export function NotificationAction({
   connectors,
   actionTypeRegistry,
 }: NotificationActionProps) {
+  const _isRuleAction = isRuleAction(action, actionTypeRegistry);
   const connectorType = connectorTypes.find(({ id }) => id === action.actionTypeId);
-  const connectorTypeName = connectorType?.name ?? '';
+  const registeredAction = actionTypeRegistry.get(action.actionTypeId);
 
-  const connector = connectors.find(({ id }) => id === action.id);
-  const connectorName = connector?.name ?? '';
+  const connectorTypeName = _isRuleAction
+    ? connectorType?.name ?? ''
+    : registeredAction.actionTypeTitle ?? '';
```

Comment:
since there is no "connector" for system actions, we need to determine the title based off the action properties in order to render helpful text on the rule details page.

Reply:
Can we add this as comment here? Or even better, extract the logic out to a function and explain it there.